src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -0,0 +1,168 @@
+     */
+    public boolean hasSessionOnDate(SessionDate sessionDate) {
+        requireAllNonNull(sessionDate);
+        return !endBefore(sessionDate) && occursAtDate(getSessionDate(), sessionDate, getInterval());
+    }
+
+    // THIS METHOD IS FOR SCHEDULE REMINDER TO RETRIEVE INFO ABOUT SESSION HAPPENING ON GIVEN DATE.
+    /**
+     * Returns a single non-recurring session on the sessionDate.
+     * @param sessionDate A valid sessionDate.
+     * @return Session of Recurring Session on particular sessionDate.
+     */
+    private Session onSessionDate(SessionDate sessionDate) {
+        requireAllNonNull(sessionDate);
+        assert(hasSessionOnDate(sessionDate));
+        return new Session(sessionDate, getDuration(), getSubject(), getFee());
+    }
+
+
+    /**
+     * Returns a non-recurring, single session that occurred before a given SessionDate.
+     * @param sessionDate A valid date, where a session exists on or before the date.
+     * @return A non-recurring single session, most recently past sessionDate.
+     */
+    public Session lastSessionOnOrBefore(SessionDate sessionDate) {
+        requireAllNonNull(sessionDate);
+        checkArgument(!startAfter(sessionDate), "There exists no session before.");
+        int numOfDaysBetween = getSessionDate().numOfDayTo(sessionDate);
+        LocalDate lastLocalDate = sessionDate.getDateTime()
+                .minusDays(numOfDaysBetween % interval.getValue()).toLocalDate();
+        SessionDate lastSessionDate = new SessionDate(
+                LocalDateTime.of(lastLocalDate, getSessionDate().getTime())
+                        .toString());
+        return this.onSessionDate(lastSessionDate);
+    }
+
+    // THIS METHOD IS EXPECTED TO BE USED IN FEE CALCULATION.
+    /**
+     * Returns the number of session in the span.
+     * @param inclusiveStart start of the span
+     * @param inclusiveEnd end of the span
+     * @return number of sessions occurring
+     */
+    public int numOfSessionBetween(SessionDate inclusiveStart, SessionDate inclusiveEnd) {
+        requireAllNonNull(inclusiveStart, inclusiveEnd);
+
+        SessionDate lastInSpan;
+        if (endBefore(inclusiveEnd)) {
+            lastInSpan = lastSessionDate;
+        } else {
+            lastInSpan = lastSessionOnOrBefore(inclusiveEnd).getSessionDate();
+        }
+
+        SessionDate firstInSpan;
+        if (startAfter(inclusiveStart)) {
+            firstInSpan = getSessionDate();
+        } else {
+            int daysDiff = getSessionDate().numOfDayTo(inclusiveStart) % interval.getValue();
+            int plusDays = daysDiff == 0
+                    ? 0
+                    : interval.getValue() - daysDiff;
+            LocalDateTime firstInSpanLdt = inclusiveStart.getDateTime().plusDays(plusDays);
+            firstInSpan = new SessionDate(firstInSpanLdt.toString());
+        }
+
+        return firstInSpan.numOfDayTo(lastInSpan) / interval.getValue() + 1;
```

Comment:
I'm assuming it's `(firstInSpan.numOfDayTo(lastInSpan) / interval.getValue()) + 1`. might be better to put brackets!